src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Serialization/RazorMessagePackSerializer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using MessagePack.Resolvers;
+using MessagePack;
+using Microsoft.AspNetCore.Razor.Serialization.MessagePack.Resolvers;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+
+namespace Microsoft.AspNetCore.Razor.Serialization;
+
+internal static class RazorMessagePackSerializer
+{
+    private static readonly MessagePackSerializerOptions s_options = MessagePackSerializerOptions.Standard
+        .WithResolver(CompositeResolver.Create(
+            RazorProjectInfoResolver.Instance,
+            StandardResolver.Instance));
+
+    public static byte[] Serialize<T>(T instance)
+        => MessagePackSerializer.Serialize(instance, s_options);
+
+    public static void SerializeTo<T>(T instance, IBufferWriter<byte> bufferWriter)
```

Comment:
Suggestion: A method that ends in "To" will normally have the first parameter be the object that is being acted on (i.e. the target of "to"). So, consider reordering the parameters in these APIs to put the `bufferWritter` or `stream` first.